app/controllers/single_page_subscriptions_controller.rb

```diff
@@ -35,16 +39,23 @@ def show
     redirect_to @content_item.fetch("base_path")
   rescue GdsApi::HTTPUnauthorized
     logout!
-    sign_in_and_confirm
+    sign_in_and_confirm(@topic_id)
```

Comment:
Do you want `topic_id` instead of `@topic_id` here? Or is this one of those premature memoization things you mentioned

Reply:
Yeah so _this_ was the source of the bug.

Within the `create` method `@topic_id` is set by `fetch_subscriber_list`.
Because we are arriving with a base_path, then need to go get the topic_id out of the content store.

After this point we're sticking the topic_id into a URL param and so we can use `topic_id` that just reads things off the URL, and  because we're using `fetch` it lets us throw 404 if we try to visit `:edit` or `:show` without one.

This _is_ a bit confusing. Because now `@topic_id` and `topic_id` are getting set from two different sources... I am wondering if I should rename or separate these...

Anyway for these purposes, because we're in the `:create` method here, if we switch this to `topic_id` then it calls the private method, which tries to read it off the param, which isn't there... so it throws a param error... which rescues with a 404... which causes a double rending error 🤦🏻

Reply:
I think you question is natural though, and may reveal that I've made a smell with overlapping names here...

Perahps I should change the name between topic_id we get out of content store and topic_id we start passing around as a param.

Reply:
Right yes, I see. I think it does make sense when I look at the controller as a whole rather than just this diff. Renaming would help though, then I'm happy for you to merge after that